Feature request: "Clean up" function to remove all the BrokenRecord custom layers

So, if I understand correctly, when you call `playback`, the order of events is:
1. BrokenRecord.jl inserts its custom layers into HTTP.jl
2. If the request is available in the file, load it from file. Otherwise, make the request and save it to file.
3. BrokenRecord.jl removes its custom layers from HTTP.jl

If an exception is thrown during step 2 (e.g. an error occurs when trying to parse the file), then the custom layers are not removed, which will break HTTP.jl for the remainder of the Julia session.

Is there a function that the user can call to "clean up" by removing all the BrokenRecord.jl custom layers from HTTP.jl?

Here's an example to illustrate what I mean. If for example there is an error when trying to parse input file, it breaks HTTP.jl for the rest of the Julia session, even if those HTTP.jl requests don't involve BrokenRecord.jl:

```julia
julia> import BrokenRecord

julia> import HTTP

julia> dir = mktempdir();

julia> cd(dir)

julia> BrokenRecord.configure!(; path=dir)

julia> open(joinpath(dir, "test.yml"), "w") do io
           println("this is not a correct yaml file")
       end
this is not a correct yaml file

julia> HTTP.get("https://julialang.org/");

julia> HTTP.get("https://julialang.org/");

julia> HTTP.get("https://julialang.org/");

julia> BrokenRecord.playback(() -> true, "test.yml")
ERROR: AssertionError: typeof(forward!(composer.input)) == DocumentStartEvent
Stacktrace:
 [1] compose_document(::YAML.Composer) at /Users/dilum/.julia/packages/YAML/aMmWk/src/composer.jl:49
 [2] compose(::YAML.EventStream) at /Users/dilum/.julia/packages/YAML/aMmWk/src/composer.jl:38
 [3] load(::YAML.TokenStream, ::Nothing; dicttype::Type{Dict{Any,Any}}) at /Users/dilum/.julia/packages/YAML/aMmWk/src/YAML.jl:22
 [4] load at /Users/dilum/.julia/packages/YAML/aMmWk/src/YAML.jl:22 [inlined]
 [5] #load#10 at /Users/dilum/.julia/packages/YAML/aMmWk/src/YAML.jl:40 [inlined]
 [6] load at /Users/dilum/.julia/packages/YAML/aMmWk/src/YAML.jl:40 [inlined]
 [7] #17 at /Users/dilum/.julia/packages/YAML/aMmWk/src/YAML.jl:87 [inlined]
 [8] open(::YAML.var"#17#18"{Base.Iterators.Pairs{Union{},Union{},Tuple{},NamedTuple{(),Tuple{}}},Nothing}, ::String, ::Vararg{String,N} where N; kwargs::Base.Iterators.Pairs{Union{},Union{},Tuple{},NamedTuple{(),Tuple{}}}) at ./io.jl:325
 [9] open at ./io.jl:323 [inlined]
 [10] #load_file#16 at /Users/dilum/.julia/packages/YAML/aMmWk/src/YAML.jl:85 [inlined]
 [11] load_file at /Users/dilum/.julia/packages/YAML/aMmWk/src/YAML.jl:85 [inlined] (repeats 2 times)
 [12] load(::Type{BrokenRecord.YAMLStorage}, ::String) at /Users/dilum/.julia/packages/BrokenRecord/1lvdc/src/storage.jl:85
 [13] before(::Type{BrokenRecord.PlaybackLayer}, ::Type{T} where T, ::String) at /Users/dilum/.julia/packages/BrokenRecord/1lvdc/src/playback.jl:6
 [14] playback(::var"#3#4", ::String; ignore_headers::Array{Any,1}, ignore_query::Array{Any,1}) at /Users/dilum/.julia/packages/BrokenRecord/1lvdc/src/BrokenRecord.jl:87
 [15] playback(::Function, ::String) at /Users/dilum/.julia/packages/BrokenRecord/1lvdc/src/BrokenRecord.jl:73
 [16] top-level scope at REPL[8]:1

julia> HTTP.get("https://julialang.org/")
ERROR: No responses remaining in the data file
Stacktrace:
 [1] error(::String) at ./error.jl:33
 [2] request(::Type{BrokenRecord.PlaybackLayer{HTTP.RedirectRequest.RedirectLayer{HTTP.BasicAuthRequest.BasicAuthLayer{HTTP.MessageRequest.MessageLayer{HTTP.RetryRequest.RetryLayer{HTTP.ExceptionRequest.ExceptionLayer{HTTP.ConnectionRequest.ConnectionPoolLayer{HTTP.StreamRequest.StreamLayer{Union{}}}}}}}}}}, ::String, ::URIs.URI, ::Array{Pair{SubString{String},SubString{String}},1}, ::Array{UInt8,1}; kwargs::Base.Iterators.Pairs{Union{},Union{},Tuple{},NamedTuple{(),Tuple{}}}) at /Users/dilum/.julia/packages/BrokenRecord/1lvdc/src/playback.jl:17
 [3] request(::Type{BrokenRecord.PlaybackLayer{HTTP.RedirectRequest.RedirectLayer{HTTP.BasicAuthRequest.BasicAuthLayer{HTTP.MessageRequest.MessageLayer{HTTP.RetryRequest.RetryLayer{HTTP.ExceptionRequest.ExceptionLayer{HTTP.ConnectionRequest.ConnectionPoolLayer{HTTP.StreamRequest.StreamLayer{Union{}}}}}}}}}}, ::String, ::URIs.URI, ::Array{Pair{SubString{String},SubString{String}},1}, ::Array{UInt8,1}) at /Users/dilum/.julia/packages/BrokenRecord/1lvdc/src/playback.jl:12
 [4] request(::String, ::String, ::Array{Pair{SubString{String},SubString{String}},1}, ::Array{UInt8,1}; headers::Array{Pair{SubString{String},SubString{String}},1}, body::Array{UInt8,1}, query::Nothing, kw::Base.Iterators.Pairs{Union{},Union{},Tuple{},NamedTuple{(),Tuple{}}}) at /Users/dilum/.julia/packages/HTTP/FV5cL/src/HTTP.jl:313
 [5] request at /Users/dilum/.julia/packages/HTTP/FV5cL/src/HTTP.jl:313 [inlined] (repeats 2 times)
 [6] #get#13 at /Users/dilum/.julia/packages/HTTP/FV5cL/src/HTTP.jl:390 [inlined]
 [7] get(::String) at /Users/dilum/.julia/packages/HTTP/FV5cL/src/HTTP.jl:390
 [8] top-level scope at REPL[9]:1

julia> HTTP.get("https://julialang.org/")
ERROR: No responses remaining in the data file
Stacktrace:
 [1] error(::String) at ./error.jl:33
 [2] request(::Type{BrokenRecord.PlaybackLayer{HTTP.RedirectRequest.RedirectLayer{HTTP.BasicAuthRequest.BasicAuthLayer{HTTP.MessageRequest.MessageLayer{HTTP.RetryRequest.RetryLayer{HTTP.ExceptionRequest.ExceptionLayer{HTTP.ConnectionRequest.ConnectionPoolLayer{HTTP.StreamRequest.StreamLayer{Union{}}}}}}}}}}, ::String, ::URIs.URI, ::Array{Pair{SubString{String},SubString{String}},1}, ::Array{UInt8,1}; kwargs::Base.Iterators.Pairs{Union{},Union{},Tuple{},NamedTuple{(),Tuple{}}}) at /Users/dilum/.julia/packages/BrokenRecord/1lvdc/src/playback.jl:17
 [3] request(::Type{BrokenRecord.PlaybackLayer{HTTP.RedirectRequest.RedirectLayer{HTTP.BasicAuthRequest.BasicAuthLayer{HTTP.MessageRequest.MessageLayer{HTTP.RetryRequest.RetryLayer{HTTP.ExceptionRequest.ExceptionLayer{HTTP.ConnectionRequest.ConnectionPoolLayer{HTTP.StreamRequest.StreamLayer{Union{}}}}}}}}}}, ::String, ::URIs.URI, ::Array{Pair{SubString{String},SubString{String}},1}, ::Array{UInt8,1}) at /Users/dilum/.julia/packages/BrokenRecord/1lvdc/src/playback.jl:12
 [4] request(::String, ::String, ::Array{Pair{SubString{String},SubString{String}},1}, ::Array{UInt8,1}; headers::Array{Pair{SubString{String},SubString{String}},1}, body::Array{UInt8,1}, query::Nothing, kw::Base.Iterators.Pairs{Union{},Union{},Tuple{},NamedTuple{(),Tuple{}}}) at /Users/dilum/.julia/packages/HTTP/FV5cL/src/HTTP.jl:313
 [5] request at /Users/dilum/.julia/packages/HTTP/FV5cL/src/HTTP.jl:313 [inlined] (repeats 2 times)
 [6] #get#13 at /Users/dilum/.julia/packages/HTTP/FV5cL/src/HTTP.jl:390 [inlined]
 [7] get(::String) at /Users/dilum/.julia/packages/HTTP/FV5cL/src/HTTP.jl:390
 [8] top-level scope at REPL[9]:1

julia> HTTP.get("https://julialang.org/")
ERROR: No responses remaining in the data file
Stacktrace:
 [1] error(::String) at ./error.jl:33
 [2] request(::Type{BrokenRecord.PlaybackLayer{HTTP.RedirectRequest.RedirectLayer{HTTP.BasicAuthRequest.BasicAuthLayer{HTTP.MessageRequest.MessageLayer{HTTP.RetryRequest.RetryLayer{HTTP.ExceptionRequest.ExceptionLayer{HTTP.ConnectionRequest.ConnectionPoolLayer{HTTP.StreamRequest.StreamLayer{Union{}}}}}}}}}}, ::String, ::URIs.URI, ::Array{Pair{SubString{String},SubString{String}},1}, ::Array{UInt8,1}; kwargs::Base.Iterators.Pairs{Union{},Union{},Tuple{},NamedTuple{(),Tuple{}}}) at /Users/dilum/.julia/packages/BrokenRecord/1lvdc/src/playback.jl:17
 [3] request(::Type{BrokenRecord.PlaybackLayer{HTTP.RedirectRequest.RedirectLayer{HTTP.BasicAuthRequest.BasicAuthLayer{HTTP.MessageRequest.MessageLayer{HTTP.RetryRequest.RetryLayer{HTTP.ExceptionRequest.ExceptionLayer{HTTP.ConnectionRequest.ConnectionPoolLayer{HTTP.StreamRequest.StreamLayer{Union{}}}}}}}}}}, ::String, ::URIs.URI, ::Array{Pair{SubString{String},SubString{String}},1}, ::Array{UInt8,1}) at /Users/dilum/.julia/packages/BrokenRecord/1lvdc/src/playback.jl:12
 [4] request(::String, ::String, ::Array{Pair{SubString{String},SubString{String}},1}, ::Array{UInt8,1}; headers::Array{Pair{SubString{String},SubString{String}},1}, body::Array{UInt8,1}, query::Nothing, kw::Base.Iterators.Pairs{Union{},Union{},Tuple{},NamedTuple{(),Tuple{}}}) at /Users/dilum/.julia/packages/HTTP/FV5cL/src/HTTP.jl:313
 [5] request at /Users/dilum/.julia/packages/HTTP/FV5cL/src/HTTP.jl:313 [inlined] (repeats 2 times)
 [6] #get#13 at /Users/dilum/.julia/packages/HTTP/FV5cL/src/HTTP.jl:390 [inlined]
 [7] get(::String) at /Users/dilum/.julia/packages/HTTP/FV5cL/src/HTTP.jl:390
 [8] top-level scope at REPL[9]:1
```

--- a/src/BrokenRecord.jl
+++ b/src/BrokenRecord.jl
@@ -82,17 +82,15 @@ function playback(
         Union{}, RecordingLayer
     end
 
-    insert_default!(before_layer, custom_layer)
     storage, path = get_storage(path, DEFAULTS[:extension])
     before(custom_layer, storage, path)
-    result = try
+    insert_default!(before_layer, custom_layer)
+    return try
         f()
     finally
         remove_default!(before_layer, custom_layer)
         after(custom_layer, storage, path)
     end
-
-    return result
 end
 
 include("recording.jl")

